service/internal/cloudtags/cloudtags.go

```diff
@@ -0,0 +1,97 @@
+package cloudtags
+
+import (
+	"context"
+	"strings"
+
+	"github.com/giantswarm/k8sclient/v5/pkg/k8sclient"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apiv1alpha2 "sigs.k8s.io/cluster-api/api/v1alpha2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/giantswarm/aws-operator/pkg/label"
+)
+
+const keyCloudPrefix = "aws-tag/"
```

Comment:
is this something that is from upstream or is this something we invented? was this consulted with sig-ux and sig-product? 

Just wanted to make sure we don't add some prefix that someone won't like a few weeks later due to bad UX :)

Reply:
yeah good point. In upstream there is no such a tag, for example, AWS cloud provider uses `service.beta.kubernetes.io/aws-load-balancer-additional-resource-tags` label for tagging LBs. In [cluster API I did not find anything](https://cluster-api-aws.sigs.k8s.io/topics/consuming-existing-aws-infrastructure.html?highlight=tag#tagging-aws-resources)

Reply:
@giantswarm/sig-ux PTAL to the naming. `aws-tag/` would be the prefix for propagating tags to cloud provider resource (example `aws-tag/my-tag-key: my-value`). Anything more appropriate?

Reply:
> FMI where does aws-tag/ come from? Is that an upstream convention? Usually, there is a domain structure defined like tag.aws.io/.

@xh3b4sd let's continue here, I am fine with aligning to domain format `tag.aws.io/`

Reply:
We discussed in SIG UX and we decided to go with `tag.provider.giantswarm.io/`